frontend/server/src/test/java/org/pytorch/serve/ModelServerTest.java

```diff
@@ -1068,6 +1068,55 @@ public void testPredictionMemoryError() throws InterruptedException {
     @Test(
             alwaysRun = true,
             dependsOnMethods = {"testPredictionMemoryError"})
+    public void testSuccessBatch() throws InterruptedException {
+        int batch_size = 4
+        Channel channel = TestUtils.connect(ConnectorType.MANAGEMENT_CONNECTOR, configManager);
+        Assert.assertNotNull(channel);
+
+        TestUtils.setHttpStatus(null);
+        TestUtils.setResult(null);
+        TestUtils.setLatch(new CountDownLatch(1));
+
+        TestUtils.registerModel(channel, "noop.mar", "err_success", true, false, batch_size=batch_size);
+        TestUtils.getLatch().await();
+
+        StatusResponse status =
+                JsonUtils.GSON.fromJson(TestUtils.getResult(), StatusResponse.class);
+        Assert.assertEquals(
+                status.getStatus(),
+                "Model \"success_batch\" Version: 1.0 registered with 1 initial workers");
+
+        channel.close().sync();
+
+        channel = TestUtils.connect(ConnectorType.INFERENCE_CONNECTOR, configManager);
+        Assert.assertNotNull(channel);
+
+        TestUtils.setResult(null);
+        TestUtils.setLatch(new CountDownLatch(1));
+        TestUtils.setHttpStatus(null);
+        
+        for(int i = 0; i < batch_size ; i ++) {
+        DefaultFullHttpRequest req =
+            new DefaultFullHttpRequest(
+                   HttpVersion.HTTP_1_1, HttpMethod.POST, "/predictions/err_success");
+        // req.content().writeCharSequence("data=invalid_output", CharsetUtil.UTF_8);
+            HttpUtil.setContentLength(req, req.content().readableBytes());
+            req.headers()
+                .set(
+                        HttpHeaderNames.CONTENT_TYPE,
+                        HttpHeaderValues.APPLICATION_X_WWW_FORM_URLENCODED);
+        channel.writeAndFlush(req);
+
+        TestUtils.getLatch().await();
+        Assert.assertEquals(TestUtils.getHttpStatus(), HttpResponseStatus.ACCEPTED);
+        }
```

Comment:
Why these 4 inference requests can confirm batching is working?

Reply:
Yeah thinking about this some more it doesn't confirm anything really. In the case of the Python backend I can create a a batch size counter in the preprocess handler and pass it all the way to the post process and then validate there that my batch size was correct. Not sure how to do the same for frontend yet